app/models/template_input.rb

```diff
@@ -2,11 +2,16 @@ class TemplateInput < ActiveRecord::Base
   class ValueNotReady < ::Foreman::Exception
   end
 
-  TYPES = { :user => N_('User input'), :fact => N_('Fact value'), :variable => N_('Variable'),
+  class ValueNotPermitted < ::Foreman::Exception
+  end
+
+  TYPES = { :user => N_('User input'), :select_options => N_('User selectable options'),
```

Comment:
While I really like the select box feature, I planned to wait with advanced forms for outcome of parameters refactoring so we can potentially reuse validators, types etc. If you think this is a must for our first version and should be implemented as a part of our plugin I think that it shouldn't be a new input type since it is user input. I think we just need to extend user input to support select options, the evaluation logic should be the same (except for validation).